components/compass-runtime-agent/internal/kyma/model/types.go

```diff
@@ -57,35 +37,14 @@ type APIDefinition struct {
 	Name        string
 	Description string
 	TargetUrl   string
-	APISpec     *APISpec
 	Credentials *Credentials
 }
 
 // EventAPIDefinition contains Event API details such
 type EventAPIDefinition struct {
-	ID           string
-	Name         string
-	Description  string
-	EventAPISpec *EventAPISpec
-}
-
-// Document contains data of document stored in the Rafter
-type Document struct {
-	ID            string
-	ApplicationID string
-	Title         string
-	DisplayName   string
-	Description   string
-	Format        DocumentFormat
-	Kind          *string
-	Data          []byte
-}
-
-// APISpec contains API spec BLOB and its type
-type APISpec struct {
-	Data   []byte
-	Type   APISpecType
-	Format SpecFormat
+	ID          string
+	Name        string
+	Description string
 }
 
```

Comment:
Can we delete this type? We don't query for EventAPISpec and there is no need to process it later.

Reply:
Probably we could do that, but I would be rather defensive here. We don't fetch spec, but we modify Application CR to store information that the Application can react on event. Should not break anything, but let's not enhance the impact of the fix.